OnPageCode.js

```diff
@@ -1,219 +1,411 @@
+        const value = PM_Network_POC.getTimeValue(
+          perfResource[parameter.timeEndKey],
+          perfResource[parameter.timeStartKey]
+        );
+        if (value > 0) {
+          currentBidCall.nw.evaluated['nw_' + parameter.key] = value;
+        }
+      });
+
+      const jsonPerfResource = JSON.stringify(perfResource);
+      const raw = JSON.parse(jsonPerfResource);
+      raw.name = PM_Network_POC.removeQueryParams(perfResource.name);
+      currentBidCall.nw.raw = raw;
+      PM_Network_POC.bidCallTracker.push(currentBidCall);
 
-    PM_Network_POC.statHatParameters.forEach(parameter => {
-      const value = PM_Network_POC.getTimeValue(
-        perfResource[parameter.timeEndKey],
-        perfResource[parameter.timeStartKey]
-      );
-      if (value > 0) {
-        output.nw.evaluated['nw_' + parameter.key] = value;
+      if (processedBidsCount === bidderRequest.bids.length) {
+        PM_Network_POC.uploadTheNetworkLatencyData(output);
+        PM_Network_POC.bidCallTracker = [];
       }
-    });
-
-    output.nw.raw = perfResource;
-    PM_Network_POC.uploadTheNetworkLatencyData(output);
-  },
-
-  performNetworkAnalysis: function (timelines, uniqueReqId) {
-    let performanceResources = window?.performance?.getEntriesByType("resource");
-    let i = PM_Network_POC.lastExecutionMaxIndex;
-    for (; i < performanceResources.length; i++) {
-      let perfResource = performanceResources[i];
-      PM_Network_POC.lastExecutionMaxIndex++;
-
-      let sspConfig = PM_Network_POC.bidders.find(
-        bidder => perfResource.name.includes(bidder.searchName)
-      ) || null;
-
-      if (sspConfig) {
-        if(sspConfig.key === 'pm') {
-          const value = PM_Network_POC.getParameterByName("uniqueReqId", perfResource.name);
-          if(value == uniqueReqId) {
-            PM_Network_POC.prepareNetworkLatencyData(perfResource, sspConfig, timelines);
+    },
+
+    removeQueryParams: url => url.split('?')[0],
+
+    isPubMaticBidder: function (bidderCode) {
+      return bidderCode === 'pubmatic';
+    },
+
+    getBidderByBidRequest: function (bidderRequest) {
+      return PM_Network_POC.bidders.find(bidder => bidder.bidderCode === bidderRequest.bidderCode);
+    },
+
+    performNetworkAnalysis: function (bidderRequests, bidsReceived) {
+      const bidReceived = bidsReceived?.find(bidReceived => PM_Network_POC.isPubMaticBidder(bidReceived?.bidderCode));
+      // latency = latency || {};
+      let performanceResources = window?.performance?.getEntriesByType('resource');
+      let lastExecutionIndex = PM_Network_POC.lastExecutionMaxIndex;
+      // PM_Network_POC.lastExecutionMaxIndex = performanceResources.length;
+
+      for (let index = 0; index < bidderRequests.length; index++) {
+        const bidderRequest = bidderRequests[index];
+        let processedBidsCount = 0;
+        let sspConfig = PM_Network_POC.getBidderByBidRequest(bidderRequest);
+
+        if (!sspConfig) break;
```

Comment:
is this break statement still needed?  or.. would we want to break from this for loop as soon as PubMatic calls get processed? (since the order of incoming ssp's varies)

Reply:
This statement is removed from the latest changes code from `OnPageCode_timeout.js`  from the branch `ns/exp/uoe-8525`